wikiscrapper/WikiQuery.py

```diff
@@ -0,0 +1,333 @@
+from collections.abc import Iterable
+from dataclasses import dataclass
+from itertools import batched
+from pprint import pprint
+import datetime
+import json
+
+
+from wikiscrapper.fetchers import FETCHERS
+from wikiscrapper.helpers import (
+    DEFAULT_DURATION,
+    DEFAULT_LANGS,
+    extract_lang_name,
+    get_session,
+    GLOBAL_LIMIT,
+    URL_INFOS,
+    Verbose,
+    WIKI_LIMIT,
+    WIKI_TITLES_LIMIT,
+)
+from wikiscrapper.WikiPage import WikiPage
+
+
+def merge_wiki_pages(found):
+    """
+    Merge linked pages with different names.
+
+    We assume here that pages are correctly linked (by Wikipedia) between each other.
+
+    :param found: dict of {name: data}.
+    :return: dict of {name: data}.
+    """
+    results = {}
+    for name, content in found.items():
+        skip = False
+        for _, pagename in content["langlinks"].items():
+            if pagename in results:  # Page with that name already exists
+                skip = True
+                break
+
+        if not skip:
+            results[name] = content
+    return results
+
+
+@dataclass
+class WikiQuery:
+    # Parameters
+    targets: str | Iterable[str] = None  # Targets to seek for
+    target_langs: str | Iterable[str] = None  # Languages to seek for
+    last_updated: datetime.datetime = datetime.datetime.now()  # When was the query made
+    target_duration: int = DEFAULT_DURATION  # Number of days for contributions
+    granularity: str = "daily"  # Granularity for contributions

```

Comment:
Change to constant in helpers (`GRANULARITY`, https://github.com/Amustache/Wikipedia-Analysor/pull/39/files#diff-219bae082311a5b1c404a801280e2d0924032ca22622026e8baeb16242cbb38fR20)